File: Property_management_system/src/com/property/management/Property.java
package com.property.management;

import java.util.ArrayList;
import java.util.List;

public class Property {
    private String propertyID;
    private String address;
    private String type;
    private String status;
    private double rentAmount;
    public static List<Property> properties = new ArrayList<>();

    public Property(String propertyID, String address, String type, String status, double rentAmount) {
        this.propertyID = propertyID;
        this.address = address;
        this.type = type;
        this.status = status;
        this.rentAmount = rentAmount;
    }

    // Getters and Setters
    public String getPropertyID() {
        return propertyID;
    }

    public void setPropertyID(String propertyID) {
        this.propertyID = propertyID;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public double getRentAmount() {
        return rentAmount;
    }

    public void setRentAmount(double rentAmount) {
        this.rentAmount = rentAmount;
    }

    // Methods
    public static void addProperty(Property property) {
        properties.add(property);
        System.out.println("Property added: " + property.address);
    }

    public static void editProperty(String propertyID, String newAddress, String newType, String newStatus, double newRentAmount) {
        for (Property property : properties) {
            if (property.propertyID.equals(propertyID)) {
                property.setAddress(newAddress);
                property.setType(newType);
                property.setStatus(newStatus);
                property.setRentAmount(newRentAmount);
                System.out.println("Property updated: " + property.address);
                return;
            }
        }
        System.out.println("Property not found!");
    }

    public static void deleteProperty(String propertyID) {
        properties.removeIf(property -> property.propertyID.equals(propertyID));
        System.out.println("Property deleted: " + propertyID);
    }
}

File: src/property.py
class Property:
    properties = []

    def __init__(self, property_id, address, property_type, status, rent_amount):
        self.property_id = property_id
        self.address = address
        self.property_type = property_type
        self.status = status
        self.rent_amount = rent_amount

    @classmethod
    def add_property(cls, property):
        cls.properties.append(property)
        print(f"Property {property.address} added.")

    @classmethod
    def edit_property(cls, property_id, new_address, new_type, new_status, new_rent_amount):
        for property in cls.properties:
            if property.property_id == property_id:
                property.address = new_address
                property.property_type = new_type
                property.status = new_status
                property.rent_amount = new_rent_amount
                print(f"Property {property_id} updated.")
                return
        print("Property not found!")

    @classmethod
    def delete_property(cls, property_id):
        cls.properties = [
            property for property in cls.properties if property.property_id != property_id]
        print(f"Property {property_id} deleted.")
